### ISSUE TITLE
Proposal: `pdk new unit_test` -> `pdk new test --unit`

### ISSUE THREAD

**Issue by scotje** [puppetlabs/pdk-planning#52]
In PDK 1.13.0 we shipped a new "experimental" feature to generate unit test scaffolding for existing resources. The subcommand for this in PDK 1.13.0 is `pdk new unit_test`. At the time, we felt this subcommand naming wasn't ideal but we didn't want to delay the feature simply because we hadn't had time to workshop the name yet.

So, now that we have had some time to reflect on the naming of the command, and considering potential future functionality, we propose that the command interface be:

`pdk new test --unit <resource>`

Example:

```
$ pdk new test --unit apache::vhost
pdk (INFO): Using Ruby 2.4.5
pdk (INFO): Using Puppet 6.8.1
[✔] Examining module contents
pdk (INFO): Creating '/Users/jesse/sandbox/pdk/apache/spec/classes/vhost_spec.rb' from template.
```

In the future the `pdk new test` command may support the generation of additional test types. (E.g. `pdk new test --acceptance`) In that scenario, if `pdk new test` is invoked without a test type flag, it would prompt the user for the type of test they wanted to generate.

Community feedback is requested on this proposal, we hope to implement this for the next PDK release at the end of September.

**Comment by scotje** [comment]
We should also make sure we define behavior when this is run non-interactively.

**Comment by scotje** [comment]
This was implemented in PDK 1.14.0.
